### PR TITLE
Replace soundcloud widget with click to play button

### PR DESCRIPTION
**Pull request by cooperq**
Port of the chrome work to firefox. cc @SwartzCr @pde for review

### CODE REVIEW

File: data/socialwidgets.js [hunk 1]
```diff
@@ -214,7 +226,11 @@ function replaceButtonWithHtmlCodeAndUnblockTracker(button, tracker, html) {
     // to prevent replacing an already removed button
     if (button.parentNode !== null) {
       let codeContainer = document.createElement("div");
-      codeContainer.innerHTML = html;
+      if(typeof html == "string") {
+        codeContainer.innerHTML = html;
```
Comment (SwartzCr) [inline data/socialwidgets.js]: can you explain or maybe put a comment in explaining this? (and the following lines)

I'm not 100% what innerHTML and outerHTML is doing

File: lib/socialWidgetHandler.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ exports.getSocialWidgetContentScriptData = function(tab) {
   socialWidgetList.forEach(function(socialWidget) {
     let socialWidgetName = socialWidget.name;
 
-    if (storage["userGreen"].hasOwnProperty(socialWidget.domain)){
+    if (storage.userGreen.hasOwnProperty(socialWidget.domain)){
```
Comment (SwartzCr) [inline lib/socialWidgetHandler.js]: when did this change? are there other places which it's broken?

Comment (cooperq) [inline lib/socialWidgetHandler.js]: I changed it because jshint was complaining about it, functionally it does the exact same thing.

**Comment by SwartzCr** [comment]
lgtm +1 to merge

**Pull request merged**